lib/orbit/evented.js

```diff
@@ -191,72 +191,36 @@ var Evented = {
       return listeners;
     },
 
-    resolve: function(eventNames) {
-      var listeners = this.listeners(eventNames),
-          args = Array.prototype.slice.call(arguments, 1);
-
-      return new Orbit.Promise(function(resolve, reject) {
-        var resolveEach = function() {
-          if (listeners.length === 0) {
-            reject();
-          } else {
-            var listener = listeners.shift();
-            var response = listener[0].apply(listener[1], args);
-
-            if (response) {
-              response.then(
-                function(success) {
-                  resolve(success);
-                },
-                function(error) {
-                  resolveEach();
-                }
-              );
-            } else {
-              resolveEach();
-            }
-          }
-        };
-
-        resolveEach();
-      });
+    resolve(eventNames, ...args) {
+      const listeners = this.listeners(eventNames);
+
+      function resolveEach([listener, ...listeners]) {
+        if(!listener) return;
```

Comment:
nit: space after `if` and please use `{\n\treturn;\n}` (unless @dgeb doesn't care about this one, I think it's the minifiers job to remove these and we should *always* include the curlies on ifs